models/repo/setting.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repo
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"code.gitea.io/gitea/models/db"
+
+	"xorm.io/builder"
+)
+
+// Setting is a key value store of repo settings
+type Setting struct {
+	ID           int64  `xorm:"pk autoincr"`
+	RepoID       int64  `xorm:"index unique(key_repoid)"`              // to load all of someone's settings
+	SettingKey   string `xorm:"varchar(255) index unique(key_repoid)"` // ensure key is always lowercase
+	SettingValue string `xorm:"text"`
+}
+
+// TableName sets the table name for the settings struct
+func (s *Setting) TableName() string {
+	return "repo_setting"
+}
+
+func init() {
+	db.RegisterModel(new(Setting))
+}
+
+// GetRepoSettings returns specific settings from repo
+func GetRepoSettings(rid int64, keys []string) (map[string]*Setting, error) {
+	settings := make([]*Setting, 0, len(keys))
+	if err := db.GetEngine(db.DefaultContext).
+		Where("repo_id=?", rid).
+		And(builder.In("setting_key", keys)).
+		Find(&settings); err != nil {
+		return nil, err
+	}
+	settingsMap := make(map[string]*Setting)
+	for _, s := range settings {
+		settingsMap[s.SettingKey] = s
+	}
+	return settingsMap, nil
+}
+
+// GetRepoAllSettings returns all settings from repo
+func GetRepoAllSettings(rid int64) (map[string]*Setting, error) {
```

Comment:
Please follow https://github.com/go-gitea/gitea/pull/16834 

How do you read `GetAllRepoSettings`? Get all repositories' settings?

`GetRepoAllSettings` means get the repository's all settings.

Reply:
And IMO this function is not handy. Think about a real case, we want to show all settings on the repository's setting page. We need a `map[string]string`, instead of `map[string]*Setting`.

And I can not imagine a real use case for `GetRepoSettings` to get the values of only a few keys.

When reviewing #16834 , I am not sure whether it's designed to be used in some cases, now it seems that there is no such case.

Reply:
I think this is to be compatible with the approach chosen for the user-specific data.
There, the same method exists as well.